### DISCUSSION TITLE
Configuring debugger layout?

### DISCUSSION THREAD

**Discussion by barrycarr** [gitpod-io/openvscode-server#421] [Q&A] [answered]
Hi, 

We'd like to provide our users with a specifically configured debugger layout (hiding some panels and expanding others). What is the best way to achieve this? 

I've figured out that the users settings default configuration values are stored in the browser IndexDB and I've found the values in there we'd like to change. But, browsing through the source for openvscode-server it's not obvious where or how the layout configuration is loaded. I've also looked and tried the approach mentioned in #199, but that doesn't seem to work for these settings. I've also looked at #255 - is this the approach I need to take here? Can I set any openvscode-server setting via `IWorkbenchConstructionOptions.configurationDefaults` as mentioned in #255 or, is there a better way to change these layout values?

Cheers

**Comment by jeanp413** [comment, marked as answer]
Hi @barrycarr, if you want all users to have a custom layout/settings as default then yes `IWorkbenchConstructionOptions.configurationDefaults` is the way to go, that's how Gitpod and codespaces also do it
You can build vscode again with `yarn gulp vscode-reh-web-linux-x64-min`

**Reply by barrycarr** [reply]
@jeanp413  Thanks very much for your answer. Cheers!
